cmds/core/printf/printf_test.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestPrintfBasic(t *testing.T) {
+
+	printf := func(args ...string) (string, string) {
+		o := &bytes.Buffer{}
+		e := &bytes.Buffer{}
+		NewPrinter(o, e, args).run()
+		return o.String(), e.String()
+	}
+
+	args := func(args ...string) []string {
+		return args
+	}
+
+	type testCase struct {
+		a   []string
+		e   string
+		err string
```

Comment:
if you have run return an error, you can use wrapper errors, and the err here can be an error type. That is preferred for tests now.

Reply:
you can use errors.Is.